snow/networking/handler/health.go

```diff
@@ -54,5 +57,27 @@ func (h *handler) networkHealthCheck() (interface{}, error) {
 		)
 	}
 
+	disconnectedVdrs := h.getDisconnectedValidators()
+	if disconnectedVdrs.Len() > 0 {
+		details["disconnectedValidators"] = disconnectedVdrs
+	}
+
 	return details, err
 }
+
+func (h *handler) getDisconnectedValidators() set.Set[ids.NodeID] {
+	allVdrs := h.peerTracker.GetValidators()
+	if allVdrs.Len() == 0 {
+		return nil
+	}
+	connectedVdrs := h.peerTracker.ConnectedValidators()
+	if connectedVdrs.Len() == 0 {
+		return allVdrs
+	} else if connectedVdrs.Len() == allVdrs.Len() {
+		return nil
+	}
+	// difference is disconnected validator
+	allVdrs.Difference(connectedVdrs)
+
+	return allVdrs
```

Comment:
I think the only one of these optimizations that could actually reduce iterations is the `if connectedVdrs.Len() == allVdrs.Len()` check, everything else wouldn't actually have any iterations anyways...

I'd imagine for any large network that there will almost always be a disconnected validator... So I feel like just having more readable code here would be best.


```suggestion
	vdrs := h.peerTracker.GetValidators()
	connectedVdrs := h.peerTracker.ConnectedValidators()
	// vdrs - connectedVdrs is equal to the disconnectedVdrs
	vdrs.Difference(connectedVdrs)
	return vdrs
```